### PR TITLE
Test Apache ORC 1.9.3 RC0

### PR DESCRIPTION
**Pull request by wgtmac**

### CODE REVIEW

**Comment by wgtmac** [comment]
delta-conversion-scala-2-12-tests (https://github.com/wgtmac/iceberg/actions/runs/8313226596/job/22763621663?pr=8) consistently fails with following error:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':iceberg-spark:iceberg-spark-3.5_2.12:compileScala'.
> Could not resolve all files for configuration ':iceberg-spark:iceberg-spark-3.5_2.12:compileClasspath'.
   > Could not find orc-core-1.9.3-shaded-protobuf.jar (org.apache.orc:orc-core:1.9.3).
     Searched in the following locations:
         file:/home/runner/.m2/repository/org/apache/orc/orc-core/1.9.3/orc-core-1.9.3-shaded-protobuf.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org/

BUILD FAILED in 46s
```

I have double checked changes in .github/workflows/delta-conversion-ci.yml and cannot figure out why. Could you help please? @dongjoon-hyun 

BTW, I have tested both main and 1.4.x branch from Apache Iceberg and they failed with different reasons.

**Comment by dongjoon-hyun** [comment]
This is the same.

- https://github.com/wgtmac/iceberg/pull/10#issuecomment-2002831352

File: .github/workflows/api-binary-compatibility.yml [hunk 1]
```diff
@@ -52,6 +52,11 @@ jobs:
           java-version: 11
       - run: |
           echo "Using the old version tag, as per git describe, of $(git describe)";
+      - run: |
```
Comment (wgtmac) [inline .github/workflows/api-binary-compatibility.yml:55]: These lines fix the api-binary check.

File: .github/workflows/delta-conversion-ci.yml [hunk 2]
```diff
@@ -86,6 +86,11 @@ jobs:
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
           restore-keys: ${{ runner.os }}-gradle-
       - run: echo -e "$(ip addr show eth0 | grep "inet\b" | awk '{print $2}' | cut -d/ -f1)\t$(hostname -f) $(hostname -s)" | sudo tee -a /etc/hosts
+      - run: |
+          mvn dependency:get -DgroupId=org.apache.orc -DartifactId=orc-core -Dversion=1.9.3 -Dpackaging=jar -D=remoteRepositories=https://repository.apache.org/content/repositories/orgapacheorc-1078
+          mvn dependency:get -DgroupId=org.apache.orc -DartifactId=orc-core -Dversion=1.9.3 -Dpackaging=jar -D=remoteRepositories=https://repository.apache.org/content/repositories/orgapacheorc-1078 -Dclassifier=nohive
+          mvn dependency:get -DgroupId=org.apache.orc -DartifactId=orc-tools -Dversion=1.9.3 -Dpackaging=jar -D=remoteRepositories=https://repository.apache.org/content/repositories/orgapacheorc-1078
+          mvn dependency:get -DgroupId=org.apache.orc -DartifactId=orc-core -Dversion=1.9.3 -Dpackaging=jar -D=remoteRepositories=https://repository.apache.org/content/repositories/orgapacheorc-1078 -Dclassifier=shaded-protobuf
```
Comment (wgtmac) [inline .github/workflows/delta-conversion-ci.yml:93]: This line fixes the shaded-protobuf issue.

**Comment by wgtmac** [comment]
I've fixed all the issues. Let me proceed to vote. @dongjoon-hyun

**Pull request closed without merging**